dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/common/hibernate/HibernateIdentifiableObjectStore.java

```diff
@@ -269,24 +269,23 @@ public void update( T object, User user )
     {
         String username = user != null ? user.getUsername() : "system-process";
 
-        if ( IdentifiableObject.class.isInstance( object ) )
+        if ( object != null )
         {
             object.setAutoFields();
 
-            BaseIdentifiableObject identifiableObject = object;
-            identifiableObject.setAutoFields();
-            identifiableObject.setLastUpdatedBy( user );
+            object.setAutoFields();
+            object.setLastUpdatedBy( user );
 
-            if ( identifiableObject.getUser() == null )
+            if ( object.getUser() == null )
             {
-                identifiableObject.setUser( user );
+                object.setUser( user );
             }
         }
 
         if ( !isUpdateAllowed( object, user ) )
         {
             AuditLogUtil.infoWrapper( log, username, object, AuditLogUtil.ACTION_UPDATE_DENIED );
-            throw new UpdateAccessDeniedException( object.toString() );
+            throw new UpdateAccessDeniedException( String.valueOf( object ) );
         }
```

Comment:
Out of curiosity, could you explain a little about this change. Since this affects a lot of stuff outside tracker also it would be interesting to know.

Reply:
Sure:

- `if ( IdentifiableObject.class.isInstance( object ) )` is replaced with `object != null` (because <T> extends `BaseIdentifiableObject`) so checking if isInstance is useless
- The rest of the change is just removing the useless cast and replace with the actual function parameter (`object`)